blockstore/apps/search/tests/test_views.py

```diff
@@ -0,0 +1,229 @@
+
+from ..core.tests.test_base import ViewsBaseTestCase
+from ..documents import BlockDocument
+from ..fieldsets import (
+    SummaryFieldSet,
+)
+
+
+class BlockDocumentsViewSetTestCase(ViewsBaseTestCase):
+    """ Tests for BlockDocumentsViewSet. """
+
+    document_class = BlockDocument
+
+    list_path = 'api:v1:index-blocks-list'
+    detail_path = 'api:v1:index-blocks-detail'
+
+    def test_list(self):
+
+        response = self.response(self.list_path)
+        assert response.data['results'] == []
+
+        titles = {'Video 1', 'Video 2', 'Video 3'}
+        for title in titles:
+            document = BlockDocument(summary=SummaryFieldSet.Document(title=title))
+            document.save(refresh=True)
+
+        response = self.response(self.list_path)
+        assert len(response.data) == 3
+        self.assertSetEqual(titles, {document['summary']['title'] for document in response.data['results']})
+
+    def test_get(self):
+
+        document = BlockDocument(summary=SummaryFieldSet.Document(title='Block to get'))
+        document.save(refresh=True)
+
+        assert BlockDocument.get(document.uuid).summary.title == 'Block to get'
+
+        response = self.response(self.detail_path, kwargs={'uuid': document.uuid})
+        self.assertDictEqual(response.data, {
+            'analytics': {'favorites': None, 'remixes': None, 'views': None},
+            'authorship': {'author_ids': []},
+            'entity': {'type': None, 'id': None},
+            'ownership': {'org_id': None},
+            'summary': {'title': 'Block to get', 'description': None, 'image': None},
+            'tags': {'paths': []},
+            'url': 'http://testserver/api/v1/index/blocks/{}/'.format(document.uuid),
+            'uuid': str(document.uuid),
+        })
+
+    def test_create(self):
+
+        response = self.response(self.list_path)
+        assert response.data['results'] == []
+
+        response = self.response(
+            self.list_path,
+            method='post',
+            data={
+                'summary': {
+                    'title': 'Block created',
+                },
+            },
+            format='json',
+            expected_response_code=201,
+        )
+
+        document = BlockDocument.get(response.data['uuid'])
+        assert document.summary.title == 'Block created'
+
+    def test_update(self):
+
+        self.maxDiff = None
+
+        document = BlockDocument(summary=SummaryFieldSet.Document(title='Block to update'))
+        document.save(refresh=True)
+
+        assert BlockDocument.get(document.uuid).summary.title == 'Block to update'
+
+        # Add some data
+        self.response(
+            self.detail_path,
+            method='patch',
+            kwargs={'uuid': str(document.uuid)},
+            data={
+                'analytics': {'favorites': 3},
+                'authorship': {'author_ids': ['The only author']},
+                'entity': {'type': 'video', 'id': 'block-v1:uuuuiidd'},
```

Comment:
There are two ways you could be using `type` here, and we need to clarify. One is metadata, just an informative piece of information for the frontend which says "this content is a video". I think that's what we want. If that's what you mean here, that's fine.

The other is as a namespace for IDs. In that case, the `type` should not be `video` because opaque keys like `block-v1:...`, `course-v1:...`, `gblock-v1`, etc. all share the same namespace.

Reply:
Sorry for the misleading example again. This is supposed to be a namespace for IDs. I have updated this to `'type': 'xblock'`.